src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs

```diff
@@ -743,44 +747,50 @@ private static void OutputPortViewModel_PropertyChanged(object sender, System.Co
         /// <param name="currentFlow"></param>
         internal static void CreateNode(Step stepInfo, StepUIAutomation uiAutomationData, bool enableFunction, GuideFlow currentFlow)
         {
-            //Node name that is expected to be created
-            var nodeCreationName = (string)uiAutomationData.JSParameters.FirstOrDefault();
-
-            //The action that will be triggered when the node is created
-            Action<NodeModel> func = (nodeModel) =>
-            {
-                GuideFlowEvents_GuidedTourNodeCreated(nodeModel, nodeCreationName, uiAutomationData.NodePosition);
-            };
+            CurrentExecutingStep = stepInfo;
 
             //If any backward action is triggered, the created needs to be deleted 
             if (currentFlow == GuideFlow.BACKWARD && lastCreatedNode != null)
             {
                 var stepMainWindow = CurrentExecutingStep.MainWindow as Window;
+
+                foreach (var connector in lastCreatedNode.AllConnectors.ToList())
+                {
+                    connector.Delete();
+                }
+
```

Comment:
It's strange that you need to do this dispose specifically, I would expect the existing API on node or workspace level to handle it, I will also take a look at this

Reply:
I see now, there is API on NodelViewModel level but not on NodeModel level